ros/src/common/libs/state_machine_lib/src/state_context.cpp

```diff
@@ -23,358 +23,334 @@
 
-bool StateContext::setCallbackUpdateFunc(const uint64_t &_state_num, const std::function<void(void)> &_f)
-{
-  bool ret = false;
-  if (getStateObject(_state_num))
-  {
-    getStateObject(_state_num)->setCallbackUpdateFunc(_f);
-    ret = true;
-  }
+      if (isCurrentState(_target_state_name))
+      {
+        return;
+      }
 
-  return ret;
-}
-bool StateContext::setCallbackInFunc(const uint64_t &_state_num, const std::function<void(void)> &_f)
-{
-  bool ret = false;
-  if (getStateObject(_state_num))
-  {
-    getStateObject(_state_num)->setCallbackInFunc(_f);
-    ret = true;
-  }
-  return ret;
-}
-bool StateContext::setCallbackOutFunc(const uint64_t &_state_num, const std::function<void(void)> &_f)
-{
-  bool ret = false;
-  if (getStateObject(_state_num))
-  {
-    getStateObject(_state_num)->setCallbackOutFunc(_f);
-    ret = true;
+      if (state_map_[transition_state_id]->getParent())
+      {
+        DEBUG_PRINT("[Child]:TransitionState: %d -> %d\n", state->getStateID(), transition_state_id);
+
+        std::shared_ptr<State> in_state = root_state_;
+
+        do
+        {
+          if (in_state == state_map_[transition_state_id]->getParent())
+          {
+            if (in_state->getChild())
+            {
+              key_list_.push_back(in_state->getChild()->getEnteredKey());
+              in_state->getChild()->onExit();
+            }
+            in_state->setChild(state_map_[transition_state_id]);
+            break;
+          }
+          in_state = in_state->getChild();
+        } while (in_state);
+
+        createDOTGraph(dot_output_name);
+        state_map_[transition_state_id]->setEnteredKey(transition_key);
+        state_map_[transition_state_id]->onEntry();
+      }
+      else
+      {
+        DEBUG_PRINT("[Root]:TransitionState: %d -> %d\n", state->getStateID(), transition_state_id);
+        state->onExit();
+
+        root_state_ = state_map_[transition_state_id];
+        root_state_->setChild(nullptr);
+        root_state_->setParent(nullptr);
+        root_state_->setEnteredKey(transition_key);
+        createDOTGraph(dot_output_name);
+
+        root_state_->onEntry();
+      }
+      break;
+    }
+    state = state->getChild();
   }
-  return ret;
-}
 
-std::string StateContext::getStateName(const uint64_t &_state_num)
-{
-  if (getStateObject(_state_num))
+  if (isCurrentState(_target_state_name))
   {
-    return getStateObject(_state_num)->getStateName();
+    showStateName();
   }
-  else
-    return "";
 }
 
-uint64_t StateContext::getStateNum(BaseState *_state)
+/*****************************/
+/* Getter/Setter             */
+/*****************************/
+
+void StateContext::createGraphTransitionList(std::ofstream& outputfile, int idx,
+                                             std::map<uint64_t, std::vector<uint64_t>>& sublist)
 {
-  if (_state)
+  /* FIX THIS
+   * Not support single state, which state is integrated before state circle.....;(
```

Comment:
Future tasks should be specified in the function of pull request or managed as isue.

Reply:
@yk-fujii 
Is this a required function?
Is it necessary to make an issue?

Reply:
I removed this because it is not needed.